src/middleware/mod.rs

```diff
@@ -10,23 +10,23 @@ pub use self::default_headers::DefaultHeaders;
 pub trait Middleware<Data>: Send + Sync {
     /// Asynchronously transform the incoming request, or abort further handling by immediately
     /// returning a response.
-    fn request(
-        &self,
-        data: &mut Data,
-        req: Request,
-        params: &RouteMatch<'_>,
-    ) -> FutureObj<'static, Result<Request, Response>> {
-        FutureObj::new(Box::new(async { Ok(req) }))
+    fn request<'a>(
+        &'a self,
+        data: &'a mut Data,
+        req: &'a mut Request,
+        params: &'a RouteMatch<'_>,
+    ) -> FutureObj<'a, Option<Response>> {
```

Comment:
We may want to use `Result<(), Response>` if we make this change, to integrate better with `?`

Reply:
That makes sense! I'll change this right now.